File: 3931-process-string-with-special-operations-i/process-string-with-special-operations-i.js
/**
 * @param {string} s
 * @return {string}
 */
var processStr = function(s) {
    let ans="";
    for(let c of s){
        if(/[a-z]/.test(c))
            ans+=c;
        else if(ans.length>0 && c=='*')
            ans=ans.slice(0,-1);
        else if(c=='#')
            ans=ans+ans;
        else if(c=='%')
            ans=ans.split('').reverse().join('');
    }
    return ans;
};

File: 3931-process-string-with-special-operations-i/process-string-with-special-operations-i.cpp
class Solution {
public:
    string processStr(string s) {
        string ans="";
        for(auto it:s){
            if(isalpha(it))
                ans+=it;
            else if(!ans.empty() && it=='*')
                ans.pop_back();
            else if(it=='#')
                ans=ans+ans;
            else if(it=='%')
                reverse(ans.begin(),ans.end());
        }
        return ans;
    }
};